sycl/plugins/cuda/pi_cuda.cpp

```diff
@@ -390,6 +390,37 @@ void getUSMHostOrDevicePtr(PtrT usm_ptr, CUmemorytype *out_mem_type,
   }
 }
 
+bool getMaxRegistersJitOptionValue(const std::string &build_options,
+                                   unsigned int &value) {
+  using namespace std::string_view_literals;
+  const std::size_t optionPos = build_options.find_first_of("maxrregcount"sv);
+  if (optionPos == std::string::npos) {
+    return false;
+  }
+
+  bool valid = false;
+  const std::size_t delimPos = build_options.find('=', optionPos + 1u);
+  const std::size_t startPos = delimPos + 1;
+  if (delimPos != std::string::npos && startPos < build_options.length()) {
```

Comment:
Nit; personally I'd prefer just doing an early exit here instead. Not blocking though.